04_health/neonatal_health_code_2022.do

```diff
@@ -0,0 +1,701 @@
+
+save "${health_data}neonatal_health_intermediate_momed_`y2'.dta", replace
+
+//# Append Education to Race Ethnicity - Keep the raceeth name the rest of the way
+use "${health_data}neonatal_health_intermediate_raceth_`y2'.dta", clear
+	append using "${health_data}neonatal_health_intermediate_momed_`y2'.dta"
+save "${health_data}neonatal_health_intermediate_raceth_`y2'.dta", replace
+
+
+//# * (5) assign "unidentified county" values to counties with missing values
+
+/// all births
+use "${health_data}neonatal_health_intermediate_all_`y2'.dta", clear
+
+* generate flag for "unidentified county"
+gen unidentified_county_flag = .
+	replace unidentified_county_flag = 1 if _merge == 2		// observations only in CDC WONDER data and not in crosswalk (unidentified counties)
+	label var unidentified_county_flag "indicator that data represent pooled unidentified counties in state"
+	drop _merge
+
+* test flag	
+assert missing(lbw_births) 		if unidentified_county_flag == 1
+assert missing(nomiss_births) 	if unidentified_county_flag == 1
+assert !missing(lbw_births) 	if missing(unidentified_county_flag)
+assert !missing(nomiss_births) 	if missing(unidentified_county_flag)
+
+* assign unidentified county value to missing counties: lbw births
+sort state county
+by state: gen unidentified_lbw = lbw_births if county == 999		//create new variable with unidentified counties value
+
+forvalues i = 1/255 {
+	by state: replace unidentified_lbw = unidentified_lbw[_n+1] if missing(unidentified_lbw)
+}
+//loop replaces the value of new variable with the value from the unidentified counties within a state
+//repeated 255 times because that is the maximum number of counties in a state
+//end result is a single variable with unidentified counties value for every observation											
+
+replace lbw_births = unidentified_lbw if unidentified_county_flag == 1  //replaces all_birth with value from unidentified counties if an unidentified county
+
+assert !missing(lbw_births) if unidentified_county_flag == 1		//test to confirm no missing values for unidentified counties
+	drop unidentified_lbw
+
+
+* assign unidentified county value to missing counties: nomiss births
+sort state county
+by state: gen unidentified_nomiss = nomiss_births if county == 999		//create new variable with unidentified counties value
+
+forvalues i = 1/255 {
+	by state: replace unidentified_nomiss = unidentified_nomiss[_n+1] if missing(unidentified_nomiss)
+}
+//loop replaces the value of new variable with the value from the unidentified counties within a state
+//repeated 255 times because that is the maximum number of counties in a state
+//end result is a single variable with unidentified counties value for every observation											
+
+replace nomiss_births = unidentified_nomiss if unidentified_county_flag == 1  //replaces all_birth with value from unidentified counties if an unidentified county
```

Comment:
should comment here read "replaces nomiss_births..." rather than "replaces all_birth"...?

Reply:
Fixed